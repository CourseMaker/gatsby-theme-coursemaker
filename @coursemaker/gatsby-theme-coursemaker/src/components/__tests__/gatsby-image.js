import React from "react"
import renderer from "react-test-renderer"
import Img from "gatsby-image"
import imageDouble from "../__mocks__/img-double";

// describe("Img", () => {
//     it("renders local correctly", () => {
//         // Created using the query from Header.js
//         const imgTestSample = imageDouble;
//         const tree = renderer.create(
//             <Img
//                 className="object-cover w-full h-40 md:h-64 md:w-1/2"
//                 fluid={imgTestSample}
//                 alt="cover image"
//                 imgStyle={{ objectPosition: "center", objectFit: "contain" }}
//             />
//         ).toJSON()
//         expect(tree).toMatchSnapshot()
//     })
// })

describe("Img", () => {
    it("renders cms correctly", () => {
        // Created using the query from Header.js
        const imgTestSample = {src: "https://coursemaker-root-bucket-dev.s3.amazonaws.câ€¦g-image.ebook_cover.png2020-10-07T20:25:12.483754"}
        const tree = renderer.create(
            <Img
                className="object-cover w-full h-40 md:h-64 md:w-1/2"
                fluid={imgTestSample}
                alt="cover image"
                imgStyle={{ objectPosition: "center", objectFit: "contain" }}
            />
        ).toJSON()
        expect(tree).toMatchSnapshot()
    })
})

