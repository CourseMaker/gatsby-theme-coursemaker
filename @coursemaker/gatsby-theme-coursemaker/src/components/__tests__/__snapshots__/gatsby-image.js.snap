// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Img renders cms correctly 1`] = `
<div
  className="object-cover w-full h-40 md:h-64 md:w-1/2 gatsby-image-wrapper"
  style={
    Object {
      "maxHeight": null,
      "maxWidth": null,
      "overflow": "hidden",
      "position": "relative",
    }
  }
>
  <div
    aria-hidden={true}
    style={
      Object {
        "paddingBottom": "NaN%",
        "width": "100%",
      }
    }
  />
  <picture>
    <img
      alt="cover image"
      loading="lazy"
      onLoad={[Function]}
      src="https://coursemaker-root-bucket-dev.s3.amazonaws.c…g-image.ebook_cover.png2020-10-07T20:25:12.483754"
      style={
        Object {
          "height": "100%",
          "left": 0,
          "objectFit": "contain",
          "objectPosition": "center",
          "opacity": 0,
          "position": "absolute",
          "top": 0,
          "transition": "opacity 500ms",
          "width": "100%",
        }
      }
    />
  </picture>
  <noscript
    dangerouslySetInnerHTML={
      Object {
        "__html": "<picture><source srcset=\\"undefined\\" /><img loading=\\"lazy\\" src=\\"https://coursemaker-root-bucket-dev.s3.amazonaws.c…g-image.ebook_cover.png2020-10-07T20:25:12.483754\\" alt=\\"cover image\\" style=\\"position:absolute;top:0;left:0;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center\\"/></picture>",
      }
    }
  />
</div>
`;
